prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -322,6 +328,105 @@ def create_transformers(bus):
     return pd.DataFrame(bus_pairs, columns=["from_bus_id", "to_bus_id"])
 
 
+def estimate_branch_impedance(branch, bus):
+    """Estimate branch impedance using transformer voltages or line voltage and length.
+
+    :param pandas.Series branch: data for a single branch (line or transformer). All
+        branches require 'type' attributes, lines require 'VOLTAGE' and 'length',
+        transformers require 'from_bus_id' and 'to_bus_id'.
+    :return: (*float*) -- impedance for that branch (per-unit).
+    """
+
+    def _euclidian(a, b):
+        return ((a[0] - b[0]) ** 2 + (a[1] - b[1]) ** 2) ** (1 / 2)
+
+    if branch.loc["type"] == "Transformer":
+        voltage_tuple = sorted(bus.loc[[branch.from_bus_id, branch.to_bus_id]])
+        reactance_lookup = pd.Series(const.transformer_reactance)
+        # Find the 'closest' voltage pair via Euclidian distance
+        closest_voltage_tuple = (
+            reactance_lookup.index.to_series()
+            .map(lambda x: _euclidian(x, voltage_tuple))
+            .idxmin()
+        )
+        return const.transformer_reactance[closest_voltage_tuple]
+    elif branch.loc["type"] == "Line":
+        # Calculate line length via sum of distances between adjacent coordinate pairs
+        reactance_lookup = pd.Series(const.line_reactance_per_mile)
+        # Find the closest voltage
+        closest_voltage = (reactance_lookup - branch.loc["VOLTAGE"]).abs().idxmin()
+        reactance_per_mile = reactance_lookup.loc[closest_voltage]
+        return branch.loc["length"] * reactance_per_mile
+    else:
+        raise ValueError(f"{branch.loc['type']} not a valid branch type")
+
+
+def calculate_branch_mileage(branch):
+    """Estimate distance of a line.
+
+    :param pandas.Series branch: data for a single line.
+    :return: (*float*) -- distance (in miles) for that line.
+    """
+    coordinates = branch.loc["COORDINATES"]
+    return sum([haversine(a, b) for a, b in zip(coordinates[:-1], coordinates[1:])])
```

Comment:
I think `Haversine` is expected `a` and `b` to (lat, lon) but the elements of `coordinates` are (lon, lat)

Reply:
You're right. Do you think it would make sense to correct this when we first load the line data, rather than remember to reverse them within `data_process`? I think everywhere else in our codebase we generally expect (lat, lon).

Reply:
`powersimdata/utility/distance.find_closest_neighbor` that I use in #217 uses (lon, lat). I can reverse it if you decide to modify the `COORDINATES` column

Reply:
I think that would be cleaner than having to remember that in this part of the codebase, we have (lon, lat), where everywhere else we have (lat, lon).

Reply:
Making it consistent across the code base is definitely cleaner. Just keep in mind we will need to refactor the functions and scripts where we call the corresponding functions as well.

Reply:
I think we are just talking to flip lon and lat in the `COORDINATES` column of the lines data frame

Reply:
I made https://github.com/Breakthrough-Energy/PreREISE/pull/223, which I think can go in upstream of this PR and #217 to avoid subsequent refactor PRs.